pkg/controller/direct/cloudbuild/externalresource.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudbuild
+
+import (
+	"strings"
+
+	cloudbuildpb "cloud.google.com/go/cloudbuild/apiv1/v2/cloudbuildpb"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/externalresource"
+)
+
+func NewResourceRef(gcpObj *cloudbuildpb.WorkerPool) *ResourceRef {
+	return &ResourceRef{
+		externalresource.New("https://cloudbuild.googleapis.com/v1/", gcpObj),
+	}
+}
+
+type ResourceRef struct {
+	*externalresource.ExternalResourceReference
+}
+
+func (e *ResourceRef) GetResourceID() string {
+	segments := strings.Split(*e.Get(), "/workerPools/")
```

Comment:
Two nits:

1) Dereferencing pointers is always scary (because it panics on nil), better to use a helper like ValueOf
2)  I think you should also check that segments actually has len != 0.

It might be the fluent way to do this is to move the parsing to NewResourceRef (which can then return an error)

Reply:
Thanks! The code looks much nicer now.